services/bettersocial/fcmToken/sendChatNotification.js

```diff
@@ -0,0 +1,20 @@
+const {messaging} = require('firebase-admin');
+const FcmTokenFunction = require('../../../databases/functions/fcmToken');
+const {FcmToken, User} = require('../../../databases/models');
+const UsersFunction = require('../../../databases/functions/users');
+
+const sendChatNotification = async (userTargetId, payload, options = {}) => {
+  const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
+  const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, signedUser);
+  if (userTargetToken) {
+    try {
+      messaging()
+        .sendToDevice(userTargetToken?.token, payload, options)
+        .then(() => {});
+    } catch (error) {
+      console.log('error send chat notifications', error);
+    }
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sendChatNotification` function could be improved by adding error handling for the case when `findSignedUserId` or `findTokenByUserId` functions throw an error. Also, the promise returned by `messaging().sendToDevice()` is not being handled properly. If it fails, it will fail silently because there's no `.catch()` attached to handle any potential errors. It's a good practice to always handle promises either with async/await and try/catch or with `.then().catch()`. 

```diff
const sendChatNotification = async (userTargetId, payload, options = {}) => {
  try {
    const signedUser = await UsersFunction.findSignedUserId(User, userTargetId);
    const userTargetToken = await FcmTokenFunction.findTokenByUserId(FcmToken, signedUser);
    if (userTargetToken) {
      messaging()
        .sendToDevice(userTargetToken?.token, payload, options)
        .then(() => {})
+        .catch((error) => {
+          console.log('Error sending chat notification', error);
+        });
    }
  } catch (error) {
    console.log('Error fetching user or token information', error);
  }
};
```
This way, we can ensure that all possible errors are logged and can be debugged later if necessary.


<!-- This is an auto-generated comment by OSS CodeRabbit -->